MarsAdvancedTaskPart1NUnitAutomationV3/Steps/ShareSkillSteps.cs

```diff
@@ -0,0 +1,240 @@
+                shareSkillButton = driver.FindElement(By.XPath("//a[@href='/Home/ServiceListing']"));
+                shareSkillButton.Click();
+            }
+            catch (Exception ex)
+            {
+                Console.WriteLine(ex);
+            }
+        }
+
+        //To capture the pop up message
+        public void CapturePopupMessage()
+        {
+
+            Thread.Sleep(1000);
+            try
+            {
+                IWebElement messageBox = driver.FindElement(By.XPath("//div[@class='ns-box-inner']"));
+                actualMessage = messageBox.Text;
+
+                IWebElement closeMessageIcon = driver.FindElement(By.XPath("//*[@class='ns-close']"));
+                closeMessageIcon.Click();
+
+                Console.WriteLine(actualMessage);
+            }
+            catch (Exception ex)
+            {
+                Console.WriteLine(ex);
+            }
+
+        }
+        public void CaptureErrorMessage()
+        {
+            Thread.Sleep(1000);
+            try
+            {
+                IWebElement errorMessage = driver.FindElement(By.XPath("//div[@class='ui basic red prompt label transition visible']"));
+                errorMessageString = errorMessage.Text;
+                Console.WriteLine(errorMessageString);
+            }
+            catch (Exception ex)
+            {
+                Console.WriteLine(ex);
+            }
+
+        }
+        public void CaptureSubcategoryErrorMessage()
+        {
+            Thread.Sleep(1000);
+            try
+            {
+                IWebElement errorMessage = driver.FindElement(By.XPath("//div[@class='ui red basic label']"));
+                errorMessageString = errorMessage.Text;
+                Console.WriteLine(errorMessageString);
+            }
+            catch (Exception ex)
+            {
+                Console.WriteLine(ex);
+            }
+
+        }
+        public void SetSubCategoryFlag()
+        {
+            subCategoryFlag = 1;
+        }
+
+        public void AddNewShareSkill(ShareSkillsDM shareSkillDM)
+        {
+
+            shareSkillsAddComponent.AddShareSkills(shareSkillDM);
+
+            expectedMessage = "Service Listing Added successfully";
+            Assert.Pass("Service Listing Added successfully");
+
+        }
+        public void AddShareSkillWithInsufficientData(ShareSkillsDM shareSkillDM)
+        {
+
+            shareSkillsAddComponent.AddShareSkills(shareSkillDM);
+
+            CapturePopupMessage();
+
+            if (subCategoryFlag == 0)
+                CaptureErrorMessage();
+            else
+            {
+                CaptureSubcategoryErrorMessage();
+                subCategoryFlag = 0;
+            }
+
+            expectedMessage = "Please complete the form correctly.";
+
+            Assert.That(actualMessage.Equals(expectedMessage), "The service listing has been added successfully");
```

Comment:
can you please move all your asserts to helper classes like you did for language assert

Reply:
I will do it.